src/org/jitsi/impl/neomedia/codec/video/vp8/DePacketizer.java

```diff
@@ -542,20 +542,91 @@ public static int getSize(byte[] input, int offset)
         }
 
         /**
+         * Determines whether the VP8 payload specified in the buffer that is
+         * passed as an argument has a picture ID or not.
+         *
+         * @param buf the byte buffer that contains the VP8 payload.
+         * @param off the offset in the byte buffer where the VP8 payload
+         * starts.
+         * @param len the length of the VP8 payload in the byte buffer.
+         * @return true if the VP8 payload contains a picture ID, false
+         * otherwise.
+         */
+        public static boolean hasPictureId(byte[] buf, int off, int len)
+        {
+            if (!isValid(buf, off))
+                return false;
+
+            if ((buf[off] & X_BIT) == 0
+                || (buf[off+1] & I_BIT) == 0)
+                return false;
+
+            return true;
+        }
+
+        /**
+         * Determines whether the VP8 payload specified in the buffer that is
+         * passed as an argument has an extended picture ID or not.
+         *
+         * @param buf the byte buffer that contains the VP8 payload.
+         * @param off the offset in the byte buffer where the VP8 payload
+         * starts.
+         * @param len the length of the VP8 payload in the byte buffer.
+         * @return true if the VP8 payload contains an extended picture ID,
+         * false otherwise.
+         */
+        public static boolean hasExtendedPictureId(byte[] buf, int off, int len)
+        {
+            return hasPictureId(buf, off, len) && (buf[off+2] & M_BIT) != 0;
+        }
+
+        /**
+         * Determines whether the VP8 payload specified in the buffer that is
+         * passed as an argument has a KEYIDX field or not.
+         *
+         * @param buf the byte buffer that contains the VP8 payload.
+         * @param off the offset in the byte buffer where the VP8 payload
+         * starts.
+         * @param len the length of the VP8 payload in the byte buffer.
+         * @return true if the VP8 payload contains a KEYIDX, false
+         * otherwise.
+         */
+        public static boolean hasKEYIDX(byte[] buf, int off, int len)
+        {
+            return (buf[off] & X_BIT) != 0
+                && ((buf[off + 1] & T_BIT) != 0 || (buf[off + 1] & K_BIT) != 0);
```

Comment:
According to the spec "If T (above) is set to 1 but K is set to 0, the TID/Y/KEYIDX octet MUST be present, but the KEYIDX field MUST be ignored."